### PR TITLE
Added CIL-GDPCIR

### PR DESCRIPTION
**Pull request by TomAugspurger**
## Description

Updates the cil-gdpcir collections

* Copied over collections
* Fixed lon / y item_asset
* Fixed geoparquet asset

Fixes https://github.com/microsoft/PlanetaryComputerDataCatalog/issues/347.

## How Has This Been Tested?

Still working on submitting this to test.

### CODE REVIEW

**Comment by TomAugspurger** [comment]
CI failure is from https://github.com/stac-utils/stac-fastapi/issues/447 / our build environment not having a C compiler.

**Comment by geospatial-jeff** [comment]
@TomAugspurger the ciso8601 dependency is removed in 2.4.1, please give it a try!

File: datasets/cil-gdpcir/dataset.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: cil_gdpcir
+image: ${{ args.registry }}/pctasks-task-base:latest
+
+args:
+- registry
+
+code:
+  src: ${{ local.path(./cil_gdpcir.py) }}
```
Comment (lossyrob) [inline datasets/cil-gdpcir/dataset.yaml:8]: This isn't used anywhere since it's all premade stac items...you can remove this block and the code file

**Review by lossyrob — CHANGES_REQUESTED** [review]
We don't have the system to run integration tests on a dataset if that dataset has changed yet, but adding that would validate the collection and items once we do (and you can run them locally before the CI is set up). Can you add something like https://github.com/microsoft/planetary-computer-tasks/blob/main/datasets/io-land-cover/tests/test_dataset.py?

**Pull request merged**